### DISCUSSION TITLE
Github Copilot does not automatically reconnect to the internet

### DISCUSSION THREAD

**Discussion by 0xbrayo** [community/community#78136] [Copilot Conversations]
### Select Topic Area

Bug

### Body

The Copilot extension on VS Code does not automatically reconnect to the internet after temporarily losing an internet connection. You have to restart  VS Code to get it to reconnect. I wish that it automatically reconnected when a stable internet connection is found.

**Comment by danielo515** [comment]
I was about to report the same. It is a bit annoying to have to restart the whole editor

**Comment by felipeanndrade** [comment]
Maybe it could be categorized as an Issue in the copilot repository to get more attention. After losing internet conection a couple times, it turns to be annoying.

**Comment by ScottMorse** [comment]
This is extremely frustrating for anyone that ever loses connection and I would think should be a no-brainer to get fixed. I have had to restart my VSCode workspace several times a day for months now with this major tool I'm paying money to use.

**Comment by 0xbrayo** [comment]
I believe the issue is related to how vs code extensions work. There's an init function that is called when vs code starts. I believe this where it checks if there's wifi connectivity and enables completions. Unfortunately this only runs once!.If you lose connectivity after the initial function call the only way to call it again is to restart the extensions(there's a command to reload extensions only just recently learnt). There are workarounds but they come with performance hits. It's fixable but it will probably take some time from what I can tell.

**Comment by ScottMorse** [comment]
Visual Studio Code: Microsoft Product
GitHub: Microsoft Product

Something is clearly wrong with how this has been developed in general, and it is not unreasonable to expect better when this is a paid product that has been around for a while now. Programs needing to handle loss of internet connection is far from a new problem. I would have sympathy if this was one of the extensions developed fully open source by an independent developer, but not for this.

**Reply by nmendezgranton** [reply]
Hi. I feel compelled to reply here, don't know why but hey the Internet is free right?
That you pay for a service does not necessarily have to make a promise that will work in the way you need it to. Paying what we pay for Copilot doesn't have the tiniest relation to the amount of value we get out of it. Chill.
And yes, I'm here frustrated by the same issue so not that I don't have the problem.

Have a great day.

**Comment by itisman** [comment]
I am struggling with this as well. I don't want to restart the whole VSCODE once it's disconnected.
for now, I found some way to deal with it. 
In VSCODE, you can press **cmd+shift+p(ctrl+shift+p)** to open the command panel, then you can **type "Reload Window"**.
the extensions will be reloaded when you reload the window. it helps for me. hope that can works for you.

Reference:
https://stackoverflow.com/questions/42002852/how-to-restart-vscode-after-editing-extensions-config

**Reply by 0xbrayo** [reply]
You can restart the extensions only via Restart extension host

**Reply by see7e** [reply]
Thank you
